src/main/java/com/google/api/generator/gapic/model/GapicClass.java

```diff
@@ -44,8 +46,28 @@ public static GapicClass create(
     return builder().setKind(kind).setClassDefinition(classDefinition).setSamples(samples).build();
   }
 
+  public static GapicClass create(Kind kind, ClassDefinition classDefinition, String defaultHost) {
```

Comment:
It's better to have less overloaded factory methods since it loses the benefit of having builders. See [here](go/java-practices/builders#when) for details. I would remove these two methods and use builders to add `defaultHost`. We should also examine all the places that created `GapicClass` and see if we really need to add `defaultHost` every time we create `GapicClass`.

Reply:
Thank you; that makes sense. I've gone through and updated to use the Builder as well as removed it where we don't need to add the `defaultHost`.